allennlp/common/util.py

```diff
@@ -354,6 +357,107 @@ def is_lazy(iterable: Iterable[A]) -> bool:
     """
     return not isinstance(iterable, list)
 
+def parse_cuda_device(cuda_device: Union[str, int, List[int]]) -> Union[int, List[int]]:
+    """
+    Disambiguates single GPU and multiple GPU settings for cuda_device param.
+    """
+    def from_list(strings):
+        if len(strings) > 1:
+            return [int(d) for d in strings]
+        elif len(strings) == 1:
+            return int(strings[0])
+        else:
+            return -1
+
+    if isinstance(cuda_device, str):
+        return from_list(re.split(r',\s*', cuda_device))
+    elif isinstance(cuda_device, int):
+        return cuda_device
+    elif isinstance(cuda_device, list):
+        return from_list(cuda_device)
+    else:
+        # TODO(brendanr): Determine why mypy can't tell that this matches the Union.
+        return int(cuda_device)  # type: ignore
+
+class ScatterableList(list):
+    """
+    A normal list, but one that should be scattered like a tensor.
+    """
+
+    # Ensure pointers will fit in a torch.LongTensor. "64 bits ought to be enough for anybody."
+    assert sizeof(c_void_p) <= sizeof(c_int64)
+
+    def to_pointer_tensor(self) -> torch.LongTensor:
+        """
+        Converts the elements to pointers, casts them to ``int64`` and then returns them in a tensor. This cast is
+        important as ``id`` gives back unsigned integers while ``torch.LongTensor`` is signed.
```

Comment:
Ouch.  That must have been annoying to find...

Reply:
Fortunately this didn't manifest as a mysterious bug. C/C++ programming has left me with a fair degree of caution around this sort of thing.